### PR TITLE
Refactor: Workspace Provider

### PR DESCRIPTION
**Pull request by RafaDev95**
https://app.clickup.com/t/86a41a5gn

### CODE REVIEW

File: src/modules/auth/hooks/useAuthActions.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  setInvalidAccount: (isInvalid: boolean) => void;
+  singleAuthentication: (params: SingleAuthentication) => void;
+  workspaceAuthentication: (params: WorkspaceAuthentication) => void;
+  workspaceAuthenticationSingle: () => void;
+};
+
+export type IUseAuthActions = IUseAuthActionsState & IUseAuthActionHandler;
+
+const useAuthActions = (): IUseAuthActions => {
+  const [state, setState] = useState<IUseAuthActionsState>({
```
Comment (guimroque) [inline src/modules/auth/hooks/useAuthActions.ts:44]: Replace this state to an query by reactquery:
- create a new hook to consume a endpoint on API `/user/me/info` [more info](https://github.com/infinitybase/bako-safe-api/pull/193), you will create an method on service, and a hook `useUserInfo`, call on `useWorkspace`.
- move function `hasWallet` on `useAuth` to the same hook -> `useUserInfo`

File: src/modules/auth/hooks/useAuthCookies.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import { CookiesConfig, CookieName } from '@/config/cookies';
+import { AuthenticateParams } from '../services';
+
+const setAuthCookies = (params: AuthenticateParams) => {
```
Comment (guimroque) [inline src/modules/auth/hooks/useAuthCookies.ts:4]: we dont need all cookies here...keep only `ACCESS_TOKEN` and `ADDRESS`

File: src/modules/workspace/hooks/useWorkspace.tsx [hunk 3]
```diff
@@ -4,63 +4,73 @@ import { useCallback, useState } from 'react';
...
 
-  const workspaceHomeRequest = useHomeDataRequest();
+  const worksapceBalance = useGetWorkspaceBalanceRequest(
+    authDetails.userInfos?.workspace?.id,
+  );
+
+  const predicatesHomeRequest = useHomeDataRequest(
+    authDetails.userInfos?.workspace?.id,
+  );
   const userWorkspacesRequest = useUserWorkspacesRequest();
```
Comment (guimroque) [inline src/modules/workspace/hooks/useWorkspace.tsx:40]: we dont need of this info here, call this hook just on header component

File: src/layouts/dashboard/header.tsx [hunk 4]
```diff
@@ -323,14 +324,16 @@ const Header = () => {
...
   const {
-    currentWorkspace,
-    workspaceDialog,
-    userWorkspacesRequest: {
-      data: userWorkspaces,
-      refetch: refetchUserWorkspaces,
+    workspaceInfos: {
+      currentWorkspace,
+      workspaceDialog,
+      userWorkspacesRequest: {
```
Comment (guimroque) [inline src/layouts/dashboard/header.tsx:330]: instance a new hook for this info `userWorkspaces`.

Comment (guimroque) [reply src/layouts/dashboard/header.tsx:330]: On `handleCancel` and `handleClose` we dont need invalidate this request `refetchUserWorkspaces()`

**Comment by guimroque** [comment]
on hook `useGetAccountsByHardwareId` add a validation to enable request:
`const validPath = window.location.pathname === '/'` and put on enable with AND actualy logic

**Comment by guimroque** [comment]
Remove `currentWorkspaceRequest` instance of `useWorkspace`, and `await currentWorkspaceRequest.refetch()` of `handleWorkspaceSelection` fn.
We don`t need this

**Comment by guimroque** [comment]
Add a new hook to get workspace members+permissions and call just on edit members dialog

**Comment by guimroque** [comment]
Remove `useTokensStore()` and create an hook-request instanced on workspaceProvider

File: src/modules/workspace/hooks/useWorkspace.tsx [hunk 5]
```diff
@@ -93,38 +100,64 @@ const useWorkspace = () => {
...
+    currentPermissions: authDetails.userInfos.workspace?.permission,
     userWorkspacesRequest,
     workspaceDialog,
     handleWorkspaceSelection: {
       handler: handleWorkspaceSelection,
       isSelecting,
     },
     navigate,
-    workspaceHomeRequest,
+    predicatesHomeRequest,
```
Comment (guimroque) [inline src/modules/workspace/hooks/useWorkspace.tsx:146]: - change this name to `latestPredicates`

File: src/modules/workspace/hooks/useWorkspace.tsx (hunk 3 above)
Comment (RafaDev95) [reply src/modules/workspace/hooks/useWorkspace.tsx:40]: Done

File: src/layouts/dashboard/header.tsx (hunk 4 above)
Comment (RafaDev95) [reply src/layouts/dashboard/header.tsx:330]: Done

**Review by guimroque — APPROVED** [review]

**Pull request merged**